pkg/api/handlers/compat/images.go

```diff
@@ -395,9 +400,21 @@ loop: // break out of for/select infinite loop
 					writeStatusCode(http.StatusInternalServerError)
 				}
 			}
-			if err := enc.Encode(report); err != nil {
-				logrus.Warnf("Failed to json encode error %q", err.Error())
+
+			if err != nil && !progressSent {
```

Comment:
Please add a comment here elaborating on this specific case and why it needs to be done (docker compat).